core/src/test/java/feign/codec/DefaultErrorDecoderTest.java

```diff
@@ -71,4 +81,22 @@ public void retryAfterHeaderThrowsRetryableException() throws Throwable {
 
     throw errorDecoder.decode("Service#foo()", response);
   }
+
+  class FeignExceptionStatusCodeMatcher extends TypeSafeMatcher<FeignException> {
```

Comment:
Working too hard! Remove this and just do try catch, assert on code. This matcher is distractingly big for the small job it does.